vars/neoDeploy.groovy

```diff
@@ -19,33 +19,103 @@ def call(parameters = [:]) {
             error "Archive cannot be found with parameter archivePath: '${archivePath}'."
         }
 
+        def deployMode = utils.getMandatoryParameter(parameters, 'deployMode', 'MTA')
+
+        def propertiesFile
+        def warAction
+        if (deployMode == 'WAR_PROPERTIESFILE') {
+            propertiesFile = new File(utils.getMandatoryParameter(parameters, 'propertiesFile', null))
+            if (!propertiesFile.isAbsolute()) {
+                propertiesFile = new File(pwd(), propertiesFile.getPath())
+            }
+            if (!propertiesFile.exists()){
+                error "Properties file cannot be found with parameter propertiesFile: '${propertiesFile}'."
+            }
+            warAction = utils.getMandatoryParameter(parameters, 'warAction', 'deploy')
+            if (warAction != 'warAction' || warAction != 'deploy') {
+                warAction = 'deploy'
+            }
+        }
+
+        def applicationName
+        def runtime
+        def runtimeVersion
+        def vmSize
+        if (deployMode == 'WAR_PARAMS') {
+            applicationName = utils.getMandatoryParameter(parameters, 'applicationName', null)
+            runtime = utils.getMandatoryParameter(parameters, 'runtime', null)
+            runtimeVersion = utils.getMandatoryParameter(parameters, 'runtimeVersion', null)
+            vmSize = utils.getMandatoryParameter(parameters, 'vmSize', 'lite')
+            if (vmSize != 'lite' || vmSize !='pro' || vmSize != 'prem' || vmSize != 'prem-plus') {
+                vmSize = 'lite'
+            }
+            warAction = utils.getMandatoryParameter(parameters, 'warAction', 'deploy')
```

Comment:
Wouldn't it be reasonable to combine retrieval of war action here with the other one in line 34ff`?
Now we have a different behaviour. Above we check the value and apply a reasonable default in case there is some nonsense. Here we miss the check. In this by intention? Could be either done only once, inside a if(deployMode == 'WAR_PARAMS' || deployMode == "WAR_PROPERTIES") {./.} or could be refactored into a method.